application/app/helpers/files_helper.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module FilesHelper
+  include DateTimeCommon
+
+  def files_progress_data(file_status_count, title = '')
+    pending = file_status_count.pending.to_i + file_status_count.downloading.to_i
+    completed = file_status_count.success.to_i
+    cancelled = file_status_count.cancelled.to_i
+    error = file_status_count.error.to_i
+    {
+      id: SecureRandom.uuid,
+      title: title,
+      pending: pending,
+      completed: completed,
+      cancelled: cancelled,
+      error: error,
+      total: file_status_count.total
+    }
+  end
```

Comment:
I would do one of these things:
- If this need to process also uploading in `file_status_count`, include this in the code
- If this only helps views related to downloads and not uploads, I would rename the method to `download_files_progress_data` to avoid confusion